test/fixtures/plugin-no-next.js

```diff
@@ -1,5 +1,5 @@
 'use strict'
 
-module.exports = function (app, opts, next) {
+module.exports = function noNext (app, opts, next) {
```

Comment:
This is a key change. We should still support assigning a fallback name to anonymous functions. I do believe this is required to facilitate showing which plugin throws an error in Fastify.

Reply:
This change definitely still assigns a fallback name, just using a different strategy than before. Previously it'd use the name of the file the function was defined in if the function is any file's default export, and otherwise it'd use the name of the function if it is named, and as a last resort it'd use the first two lines of the functions source. Now, we use metadata as the first strategy, and then the name if it exists, and then the source. This diff is just to make the test a bit terser.

Reply:
But this change loses the test covering anonymous functions.

Reply:
@airhorns could you add a new test?

Reply:
Ok, updated to add a test, I still think this file should be changed though as it pertains to timeout tests, not plugin naming